app/helpers/sessions_helper.rb

```diff
@@ -3,15 +3,30 @@ def log_in user
     session[:user_id] = user.id
   end
 
+  def remember user
+    user.remember
+    cookies.permanent.signed[:user_id] = user.id
```

Comment:
I have a question: What is signed cookie, I meant what is the purpose of using signed cookie here?

Reply:
@mdmahmudurrahman cookie keep login infomation in client to help server detect client has logged in before when client visit website.

Reply:
@phucynwa No, I meant why you used signed here?

Reply:
@mdmahmudurrahman Yeah, it was used in tut, but I think it isn't really need in this case. User ID haven't to be encode in this case.

Reply:
No, you study about signed cookie first, it's purpose, usefulness etc.

Reply:
@mdmahmudurrahman it prevents users change their user id. right ?

Reply:
@phucynwa And this user is a malicious user/hacker who temper cookie.  =))